examples/graphs/proteins_pytorch_geometric/model_def.py

```diff
@@ -0,0 +1,149 @@
+import tempfile
+from math import ceil
+
+import torch
+import torch.nn.functional as F
+from determined.pytorch import DataLoader, PyTorchTrial, PyTorchTrialContext
+from torch.nn import Linear
+from torch.utils.data import random_split
+from torch_geometric.data.dataloader import Collater
+from torch_geometric.datasets import TUDataset
+from torch_geometric.nn import DenseGraphConv, GCNConv, dense_mincut_pool
+from torch_geometric.utils import to_dense_adj, to_dense_batch
+
+
+# Ported from https://github.com/rusty1s/pytorch_geometric/blob/master/examples/proteins_mincut_pool.py
+class Net(torch.nn.Module):
+    def __init__(self, in_channels, out_channels, hidden_channels, average_nodes):
+        super(Net, self).__init__()
+
+        self.conv1 = GCNConv(in_channels, hidden_channels)
+        num_nodes = ceil(0.5 * average_nodes)
+        self.pool1 = Linear(hidden_channels, num_nodes)
+
+        self.conv2 = DenseGraphConv(hidden_channels, hidden_channels)
+        num_nodes = ceil(0.5 * num_nodes)
+        self.pool2 = Linear(hidden_channels, num_nodes)
+
+        self.conv3 = DenseGraphConv(hidden_channels, hidden_channels)
+
+        self.lin1 = Linear(hidden_channels, hidden_channels)
+        self.lin2 = Linear(hidden_channels, out_channels)
+
+    def forward(self, x, edge_index, batch):
+        x = F.relu(self.conv1(x, edge_index))
+
+        x, mask = to_dense_batch(x, batch)
+        adj = to_dense_adj(edge_index, batch)
+
+        s = self.pool1(x)
+        x, adj, mc1, o1 = dense_mincut_pool(x, adj, s, mask)
+
+        x = F.relu(self.conv2(x, adj))
+        s = self.pool2(x)
+
+        x, adj, mc2, o2 = dense_mincut_pool(x, adj, s)
+
+        x = self.conv3(x, adj)
+
+        x = x.mean(dim=1)
+        x = F.relu(self.lin1(x))
+        x = self.lin2(x)
+        return F.log_softmax(x, dim=-1), mc1 + mc2, o1 + o2
+
+
+class GCNTrial(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext):
+        self.context = context
+
+        # Create a unique download directory for each rank so they don't overwrite each
+        # other when doing distributed training.
+        download_directory = tempfile.mkdtemp()
+
+        self.dataset = TUDataset(
+            root=download_directory,
+            name=self.context.get_hparam("dataset"),
+            use_node_attr=True,
+        )
+
+        val_ratio = 0.2
```

Comment:
Can we put this in the .yaml file config? I don't like hardcoded values like this in the model definition.